pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ControllerLeaderLocatorIntegrationTest.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.integration.tests;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.helix.model.ExternalView;
+import org.apache.helix.model.MasterSlaveSMD;
+import org.apache.pinot.common.utils.CommonConstants.Helix;
+import org.apache.pinot.common.utils.helix.LeadControllerUtils;
+import org.apache.pinot.controller.ControllerConf;
+import org.apache.pinot.controller.ControllerStarter;
+import org.apache.pinot.controller.helix.ControllerTest;
+import org.apache.pinot.pql.parsers.utils.Pair;
+import org.apache.pinot.server.realtime.ControllerLeaderLocator;
+import org.apache.pinot.util.TestUtils;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class ControllerLeaderLocatorIntegrationTest extends ControllerTest {
+  private static long TIMEOUT_IN_MS = 1000_000L;
+  private HashMap<Integer, String> _partitionToTableMap;
+
+  @BeforeClass
+  public void setUp() {
+    startZk();
+    startController();
+
+    ControllerLeaderLocator.create(_helixManager);
+
+    findTableNamesForAllPartitions();
+  }
+
+  @Test
+  public void testControllerLeaderLocator() {
+    String testTableName1 = "testTable";
```

Comment:
Can you use the pre-generated names in this test as well? Just keeps it all uniform

Reply:
Updated the code to use pre-generated names